numpy/lib/format.py

```diff
@@ -259,6 +259,43 @@ def dtype_to_descr(dtype):
     else:
         return dtype.str
 
+def descr_without_paddings(descr):
+    aligned = False
+    newdescr = {'names': [], 'formats': [], 'offsets': []}
+    offset = 0
+    unique_name = 'invisible_field_for_internal_descr_without_paddings'
+    for itr, tup in enumerate(descr):
+        if tup[0]=='':
```

Comment:
You should do this branch after calculating `fmt`, not before - the logic for calculating `fmt` is unchanged by whether the field is named.